src/SwitchScreenFloatingButton.tsx

```diff
@@ -28,7 +29,8 @@ export const SwitchScreenFloatingButton = ({
 
   const panResponder = useRef(
     PanResponder.create({
-      onMoveShouldSetPanResponder: () => true,
+      onMoveShouldSetPanResponder: (_, { dx, dy }: PanResponderGestureState) =>
+        dx > 0 || dy > 0,
```

Comment:
What happens if we move to the left or the top and `dx` or `dy` are negative ?

Reply:
`dx` and `dy` are absolute distances, as commented in the code:
```
/**
   * accumulated distance of the gesture since the touch started
   */
  dx: number;

  /**
   * accumulated distance of the gesture since the touch started
   */
  dy: number;
```